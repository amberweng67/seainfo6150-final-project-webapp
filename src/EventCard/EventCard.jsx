import React from "react";
import styles from './EventCard.module.css';

const EventCard = ({card}) => {
  return (
    <div className={styles.cardContainer}>
      <div>{card.title.toUpperCase()}</div>
      <div><span role="img" aria-label="calendar icon for time">📆</span> {card.time}</div>
      <div><span role="img" aria-label="pin icon for location">📌</span> {card.location}</div>
      <div>{card.intro}</div>
      <div><span role="img" aria-label="small icon for host">😁</span> {card.host}</div>
    </div>
  );
};

export default EventCard;