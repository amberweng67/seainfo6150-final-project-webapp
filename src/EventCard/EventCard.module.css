.cardContainer {
  height: 400px;
  width: 200px;
  background-size: cover;
  border-radius: 20px;
  padding: 20px;
  border: solid 1px #e3e6ea;
}

.cardContainer > img {
  width: 100%;
}
.cardContainer:hover{
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.3);
}
