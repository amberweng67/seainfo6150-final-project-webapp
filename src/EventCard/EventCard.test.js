import React from "react";
import EventCard from "./EventCard.jsx";

const CARD =
{
  "id": "90e2b43a-3183-50e9-b75b-3af7e4765f18",
  "title": "potatoes",
  "time": "12/4/2069",
  "location": "Slovakia",
  "intro": "volume supply poor tongue camp decide mirror turn trunk fully everything parts realize pony shelter see crop numeral clay doubt brush track great cabin",
  "host": "Larry Cohen"
};

it("renders the event card component correctly", () => {
  const { container } = render(<EventCard card={CARD} />);
  expect(container).toMatchSnapshot();
});
