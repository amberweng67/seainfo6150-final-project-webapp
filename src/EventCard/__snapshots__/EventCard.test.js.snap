// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the event card component correctly 1`] = `
<div>
  <div
    class="cardContainer"
  >
    <div>
      POTATOES
    </div>
    <div>
      <span
        aria-label="calendar icon for time"
        role="img"
      >
        ðŸ“†
      </span>
       
      12/4/2069
    </div>
    <div>
      <span
        aria-label="pin icon for location"
        role="img"
      >
        ðŸ“Œ
      </span>
       
      Slovakia
    </div>
    <div>
      volume supply poor tongue camp decide mirror turn trunk fully everything parts realize pony shelter see crop numeral clay doubt brush track great cabin
    </div>
    <div>
      Larry Cohen
    </div>
  </div>
</div>
`;
